### ISSUE TITLE
[0.4.3] restrict-plus-operands rejects valid bigint + bigint operations

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1261]
## Summary

Valid bigint additions are rejected both for bigint-annotated parameters and bigint literals. The rule should accept bigint + bigint, while still rejecting bigint + number.

Verified with @utoo/lint 0.4.3, ESLint 8.57.1, @typescript-eslint/parser and eslint-plugin 6.21.0, Node.js 22.23.1 on macOS arm64. This reproduction does not depend on different allowAny defaults between plugin versions.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.3 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0
```

`tsconfig.json`:

```json
{"compilerOptions":{"strict":true,"target":"ES2022","jsx":"react"},"include":["example.tsx"]}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "project": "./tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

Run each snippet separately as `example.tsx`:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
```

## Reproductions

```tsx
export function example(a:bigint,b:bigint){return a+b;}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string. Got `bigint`.
```

```tsx
export const result=1n+2n;
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string. Got `bigint`.
```

## Controls

```tsx
export function example(a:bigint,b:number){return a+b;}
```

ESLint:

```text
Numeric '+' operations must either be both bigints or both numbers. Got `bigint` + `number`.
```

utlint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string. Got `bigint`.
```

```tsx
export function example(a:number,b:number){return a+b;}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
(no diagnostics)
```

There are no parsing failures or lint-disable comments, and only this rule is enabled.
